### ISSUE TITLE
обход диагоналей 1

### ISSUE THREAD

**Issue by DmitriyGordon** [DmitriyGordon/iteaNoughtsAndCrosses#8]
Написать функцию которая проходит через диагонали матрицы
#### она обязательно должна адаптироваться под размер поля и количество ячеек для победы!!! Пока что данная функция возвращать ничего не будет.
![3](https://user-images.githubusercontent.com/21203839/32323371-753df7f4-bfd0-11e7-98cf-a07b1b2cd93a.png)
аргументы - поле,  размер поля, количество символов для победы
### функция не должна проходить через диагонали, количесво ячеек в которой меньше количества для победы.

**Comment by ksavchenko** [comment]
> аргументы - поле, размер поля, количество символов для победы

а зачем передавать в функцию и поле и его размер? можно передать только поле, а размер уже определять с помощью len(field)?

**Comment by DmitriyGordon** [comment]
Это не совсем правильно. В данном случае действительно можно получить тот же размер поля используя функцию len, но есть пару нюансов.
1) На этот момент у нас уже будет переменная которая хранит размер поля (может вы этого не видите, потому что до сих пор никто не написал функцию которая создаёт поле), так что зачем опять считать его размер.
2) в данном случае функция len вернёт не совсем его размер, а именно количество row в нашей матрице. Так что это работает только с квадратичной матрицей.
3) если мы решим в будущем поменять представление нашего поля, например создадим целый класс поля, придётся переопределять мэджик метод класса, что совсем нелогично, учитывая, что у нас будет атрибут класса с правильным размером.

Да я согласен, что передавать аргументом его не обязательно

**Comment by ksavchenko** [comment]
Да, я почему-то даже не подумала, что матрица в игре может быть не квадратная :(

**Comment by DmitriyGordon** [comment]
да, конечно же может, но мы будем делать квадратичной, потому что так проще и быстрее проверить победу или ничью.
если есть желание, можно в отдельную ветку делать комиты, для не квадратичной матрицы, там просто будет другой алгоритм обхода диагоналей, плюс надо будет делать проверку на возможность победы по диагонали, вертикали и горизонтали одновременно при выбранных размерах поля и количества символов для победы
